practices/code-review.md

```diff
@@ -4,44 +4,61 @@
+
 
 ## How
 
-Here's the general protocol that we use:
+Here's the general process that we use:
 
-1. Create a local short-lived feature branch based off master.
-* When feature is complete and tests pass, stage the changes and commit them.
-* Write a [good commit message](../git/git.md#commit-messages).
-* Submit a [pull request](https://help.github.com/articles/using-pull-requests/).
-* Ask for a code review in Slack.
-* A colleague (or multiple colleagues) other than the author reviews the pull request, and they make comments and ask questions directly on lines of code in the GitHub web interface.
-* When satisfied, the reviewer(s) will comment on the pull request with a +1 or some other simple message indicating that the code is ready to merge.
-* Rebase interactively. Squash commits like "Fix whitespace", or micro-commits that serve no purpose without the following commits into one or a small number of valuable commit(s). Edit commit messages to reveal intent.
-* View a list of new commits. View changed files. Merge branch into master.
-* Delete your remote feature branch.
-* Delete your local feature branch.
+1. Create a local branch based off master.
+1. Branches should adhere to the following best practices to ensure they align with [Continous Delivery](https://martinfowler.com/bliki/ContinuousDelivery.html) & [Continuous Integration](https://martinfowler.com/articles/continuousIntegration.html) as much as is reasonable (given the constraints of team structure and the discipline);
+	* Branches should be as short lived as possible.
+	* Branches should be _atomic_ i.e. one unit of work that cannot be sensibly subdivided.
+		* Therefore they should contain their own tests.
+		* Therefore they should be releasable independently.
+	* Branches should be as limited in scope as possible. Large PR's are hard to review and can be painful to merge. Also, once merged & a bug is traced back to that commit, they are much harder to debug.
+1. When you are getting close to being ready for PR, `git rebase master` -- especially if other devs are working on the same code.  Avoid rebase hell by;
+	* _talking to your team_ to divide work across files sensibly & co-ordinate merges if required.
+	* pull origin master & rebase often.
+	* make sure branches are short-lived! (Did we mention that branches should be short-lived?)
+1. When the unit of work is complete and tests pass, stage the changes and commit them.
+1. Write a [good commit message](../git/git.md#commit-messages).
+1. Submit a [pull request](https://help.github.com/articles/using-pull-requests/).
+1. Ask for a code review in Slack.
+1. A colleague (or multiple colleagues) other than the author reviews the pull request, and they make comments and ask questions directly on lines of code in the GitHub web interface.
+1. When satisfied, the reviewer(s) will comment on the pull request with a +1 or some other simple message indicating that the code is ready to merge.
```

Comment:
I think we should make this more generall "will approve the pull request" since there's explicit approvals in GitHub now (and we should be using them!)